pkg/kv/kvserver/replica_proposal.go

```diff
@@ -533,10 +533,16 @@ func (r *Replica) leasePostApplyLocked(
 	// seconds) in any case due to the liveness heartbeats. And the system config
 	// will be gossiped rarely because it falls on a range with an epoch-based
 	// range lease that is only reacquired extremely infrequently.
-	if iAmTheLeaseHolder {
+	//
+	// TODO(erikgrinaker): This and MaybeGossipNodeLivenessRaftMuLocked should
+	// check whether the replica intersects the liveness span rather than contains
+	// the entirety of it.
+	if iAmTheLeaseHolder && kvserverbase.ContainsKeyRange(
```

Comment:
If liveness accidentally split, this would make that much worse - just thinking about this because of https://github.com/cockroachdb/cockroach/issues/96365, though there we have the inverse problem.

Seems fine just couldn't help thinking about it.

Reply:
Yeah, it's not great.